aodncore/pipeline/handlerbase.py

```diff
@@ -659,16 +673,25 @@ def _store_unharvested(self):
         if files_to_store:
             self._upload_runner.run(files_to_store)
 
-    def _publish(self):
+    def _pre_publish(self):
         self.file_collection.set_archive_paths(self._archive_path_function_ref)
         self.file_collection.validate_attribute_uniqueness('archive_path')
-        self._archive()
+
+        if self.allowed_archive_path_regexes:
+            self.file_collection.validate_attribute_value_matches_regexes('archive_path',
+                                                                          self.allowed_archive_path_regexes)
```

Comment:
Shouldn't this be applied only to files that will actually be archived?
Likewise for the dest_path validation, should only be applied to files that will be stored, harvested or deleted.

Reply:
Yes it should!